sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsController.java

```diff
@@ -392,4 +402,62 @@ private Consumer<List<EventParticipantIndexDto>> bulkOperationCallback(
 			}
 		};
 	}
+
+	public void openLineListingWindow(EventReferenceDto eventReferenceDto) {
+
+		Window window = new Window(I18nProperties.getString(Strings.headingLineListing));
+
+		EventDto eventDto = FacadeProvider.getEventFacade().getByUuid(eventReferenceDto.getUuid());
+
+		LineListingLayout lineListingForm = new LineListingLayout(window, eventDto);
+
+		lineListingForm.setSaveCallback(eventParticipants -> saveEventParticipantsFromLineListing(lineListingForm, eventParticipants));
+
+		window.setContent(lineListingForm);
+		window.setModal(true);
+		window.setPositionX((int) Math.max(0, (Page.getCurrent().getBrowserWindowWidth() - lineListingForm.getWidth())) / 2);
+		window.setPositionY(70);
+		window.setResizable(false);
+
+		UI.getCurrent().addWindow(window);
+
+	}
+
+	private void saveEventParticipantsFromLineListing(LineListingLayout lineListingForm, LinkedList<LineDto<EventParticipantDto>> eventParticipants) {
+		try {
+			lineListingForm.validate();
+		} catch (ValidationRuntimeException e) {
+			Notification.show(I18nProperties.getString(Strings.errorFieldValidationFailed), "", Type.ERROR_MESSAGE);
+			return;
+		}
+
+		while (!eventParticipants.isEmpty()) {
+			LineDto<EventParticipantDto> eventParticipantLineDto = eventParticipants.pop();
+			EventParticipantDto newEventParticipant = eventParticipantLineDto.getEntity();
+			PersonDto newPerson = eventParticipantLineDto.getPerson();
+
+			ControllerProvider.getPersonController()
+				.selectOrCreatePerson(newPerson, I18nProperties.getString(Strings.infoSelectOrCreatePersonForEventParticipant), selectedPerson -> {
+					if (selectedPerson != null) {
+						if (FacadeProvider.getEventParticipantFacade().exists(selectedPerson.getUuid(), lineListingForm.getEventDto().getUuid())) {
+							throw new Validator.InvalidValueException(I18nProperties.getString(Strings.messageAlreadyEventParticipant));
+						} else {
+							newEventParticipant.setPerson(FacadeProvider.getPersonFacade().getByUuid(selectedPerson.getUuid()));
+							EventParticipantDto savedDto = eventParticipantFacade.save(newEventParticipant);
+
+							Notification notification =
+								new Notification(I18nProperties.getString(Strings.messagePersonAddedAsEventParticipant), "", Type.HUMANIZED_MESSAGE);
+							notification.show(Page.getCurrent());
+
+							Notification.show(I18nProperties.getString(Strings.messageEventParticipantCreated), Type.ASSISTIVE_NOTIFICATION);
+						}
+						if (eventParticipants.isEmpty()) {
```

Comment:
closing the dialog must be done after the `while` statement

Reply:
if the notification is moved after the while statement it will be seen before the duplicate person dialog, most likely because they run in different threads

Reply:
found another way to fix notification issue